### ISSUE TITLE
Deprecate shell inference

### ISSUE THREAD

**Issue by Schniz** [Schniz/fnm#365]
Something to consider. It's out of fnm's scope anyway. We know on what shell we're installing it into, and projects like Starship don't infer shells too.

**Comment by ismay** [comment]
If all commands would accept the `--shell=fish` argument, a user could just create this alias:

alias f="fnm --shell=fish"

(Or something like it, in their shell). Currently however, it doesn't seem like all commands accept this argument, so the alias doesn't work. It fails with fnm use for example. Or am I misunderstanding how this works?

**Comment by Schniz** [comment]
Yes, but only `env` and `completions`, which are commands that should only appear in the users' shell profile, need the `shell` argument. So I don't see why it needs to be a global config argument 😄

**Comment by ismay** [comment]
Ah I see. So I misunderstood how to use it. Ok, thanks for the explanation!

**Comment by Schniz** [comment]
After thinking about this more, fully deprecating shell inference isn't really feasible — too many users rely on it and it would be a painful breaking change.

What we can do instead is update the installation scripts to include an explicit shell value by default, and update the documentation to mention that pinning to a specific shell (instead of inferring at runtime) is faster. That way new users get the better default, and existing users can opt in.

Working on a PR for this.

---

*This message was written by AI. Sorry it's not personal, but it's helping me maintain this project. Feel free to disagree with the statement and reopen the issue if needed.*
